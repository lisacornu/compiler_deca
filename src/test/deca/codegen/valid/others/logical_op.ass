; start main program
; Main program
; Beginning of main instructions:
	WSTR "TEST AND"
	WNL
	WSTR "Devrait print : true"
	WNL
	LOAD #1, R2
	LOAD #1, R3
	LOAD R3, R0
	CMP #0, R0
	BEQ and_false_case0
	CMP #0, R2
	BEQ and_false_case0
	BRA and_end_case0
and_false_case0:
	LOAD #0, R2
and_end_case0:
	CMP #1, R2
	BNE elseStart0
	WSTR "true"
	WNL
	BRA ifThenElseExit0
elseStart0:
ifThenElseExit0:
	LOAD #0, R2
	LOAD #1, R3
	LOAD R3, R0
	CMP #0, R0
	BEQ and_false_case1
	CMP #0, R2
	BEQ and_false_case1
	BRA and_end_case1
and_false_case1:
	LOAD #0, R2
and_end_case1:
	CMP #1, R2
	BNE elseStart1
	WSTR "false"
	WNL
	BRA ifThenElseExit1
elseStart1:
ifThenElseExit1:
	LOAD #1, R2
	LOAD #0, R3
	LOAD R3, R0
	CMP #0, R0
	BEQ and_false_case2
	CMP #0, R2
	BEQ and_false_case2
	BRA and_end_case2
and_false_case2:
	LOAD #0, R2
and_end_case2:
	CMP #1, R2
	BNE elseStart2
	WSTR "false"
	WNL
	BRA ifThenElseExit2
elseStart2:
ifThenElseExit2:
	LOAD #0, R2
	LOAD #0, R3
	LOAD R3, R0
	CMP #0, R0
	BEQ and_false_case3
	CMP #0, R2
	BEQ and_false_case3
	BRA and_end_case3
and_false_case3:
	LOAD #0, R2
and_end_case3:
	CMP #1, R2
	BNE elseStart3
	WSTR "false"
	WNL
	BRA ifThenElseExit3
elseStart3:
ifThenElseExit3:
	WSTR "TEST OR"
	WNL
	WSTR "Devrait print : true true true"
	WNL
	LOAD #1, R2
	LOAD #1, R3
	LOAD R3, R0
	CMP #0, R0
	BNE or_true_case0
	CMP #0, R2
	BNE or_true_case0
	BRA or_end_case0
or_true_case0:
	LOAD #1, R2
or_end_case0:
	CMP #1, R2
	BNE elseStart4
	WSTR "true"
	WNL
	BRA ifThenElseExit4
elseStart4:
ifThenElseExit4:
	LOAD #0, R2
	LOAD #1, R3
	LOAD R3, R0
	CMP #0, R0
	BNE or_true_case1
	CMP #0, R2
	BNE or_true_case1
	BRA or_end_case1
or_true_case1:
	LOAD #1, R2
or_end_case1:
	CMP #1, R2
	BNE elseStart5
	WSTR "true"
	WNL
	BRA ifThenElseExit5
elseStart5:
ifThenElseExit5:
	LOAD #1, R2
	LOAD #0, R3
	LOAD R3, R0
	CMP #0, R0
	BNE or_true_case2
	CMP #0, R2
	BNE or_true_case2
	BRA or_end_case2
or_true_case2:
	LOAD #1, R2
or_end_case2:
	CMP #1, R2
	BNE elseStart6
	WSTR "true"
	WNL
	BRA ifThenElseExit6
elseStart6:
ifThenElseExit6:
	LOAD #0, R2
	LOAD #0, R3
	LOAD R3, R0
	CMP #0, R0
	BNE or_true_case3
	CMP #0, R2
	BNE or_true_case3
	BRA or_end_case3
or_true_case3:
	LOAD #1, R2
or_end_case3:
	CMP #1, R2
	BNE elseStart7
	WSTR "false"
	WNL
	BRA ifThenElseExit7
elseStart7:
ifThenElseExit7:
	HALT
; end main program
