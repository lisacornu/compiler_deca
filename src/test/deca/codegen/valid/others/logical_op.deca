// Description:
//    test des conditions logiques and et or
//
// Resultats:
//   ok
//
// Historique:
//    cree le 11/01/2025

{
    println("TEST AND");
    println("Devrait print : true");

    if (true && true) {
        println("true");
    }
    if (false && true) {
        println("false");
    }
    if (true && false) {
        println("false");
    }
    if (false && false) {
        println("false");
    }

    println("TEST OR");
    println("Devrait print : true true true");

    if (true || true) {
        println("true");
    }
    if (false || true) {
        println("true");
    }
    if (true || false) {
        println("true");
    }
    if (false || false) {
        println("false");
    }
}