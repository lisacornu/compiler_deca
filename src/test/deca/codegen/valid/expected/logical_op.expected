TEST AND
Devrait print : true
true
TEST OR
Devrait print : true true true
true
true
true